external/defects4c/evaluation/run_tests.py

```
@@ -0,0 +1,96 @@
+        sys.exit(1)
+    project_dir = os.path.join(os.path.abspath(args.work_dir), project)
+    if not os.path.isdir(project_dir):
+        print(f"Project dir not found: {project_dir}. Run checkout_bug.py first.", file=sys.stderr)
+        sys.exit(1)
+
+    build_cmd = (bug.get("build_cmd") or "").strip()
+    test_cmd = (bug.get("test_cmd") or "make check").strip()
+
+    if args.build_first and build_cmd:
+        print("Building:", build_cmd)
+        rc = subprocess.run(build_cmd, shell=True, cwd=project_dir)
```

Ruff (0.15.2)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)